heareval/evaluation/task_evaluation.py

```diff
@@ -10,91 +10,123 @@
-) -> Dict[str, float]:
-    #    return {"macroauc": 0.0}
-    # The rest is broken if test set vocabulary is a strict subset of train vocabulary.
-    # TODO: This should happen in task_evaluation, not be reused everywhere
-    # Dictionary of labels and integer idx: {label -> idx}
-    label_vocab = label_vocab.set_index("label").to_dict()["idx"]
-    # TODO: This shape only works for multiclass, check that is the type
-    for rowlabels in targets:
-        assert len(rowlabels) == 1
-    y_true = np.array([label_vocab[rowlabels[0]] for rowlabels in targets])
-
-    # Convert to multilabel one-hot
-    y_true_multilabel = np.zeros(predictions.shape)
-    y_true_multilabel[np.arange(y_true.size), y_true] = 1
-
-    import IPython
-
-    ipshell = IPython.embed
-    ipshell(banner1="ipshell")
-    return {"macroauc": sklearn.metrics.roc_auc_score(y_true, predictions)}
-
+class MetricFunction:
+    """
+    A simple abstract base class for metric functions
+    """
 
-def chroma_error(
-    predictions: np.ndarray, targets: List, label_vocab: pd.DataFrame
-) -> Dict[str, float]:
+    def __init__(self, task_metadata: Dict, label_vocab: pd.DataFrame):
+        self.task_metadata = task_metadata
+        assert "idx" in label_vocab.columns, "label_vocab missing idx column"
+        assert "label" in label_vocab, "label_vocab missing label column"
+        self.label_vocab = label_vocab
+
+    def label_vocab_as_dict(self, key: str) -> Dict:
+        """
+        Returns a dictionary of the label vocabulary mapping the label column to
+        the idx column. key sets whether the label or idx is the key in the dict. The
+        other column will the the value.
+        """
+        if key == "label":
+            value = "idx"
+        else:
+            assert key == "idx", "key argument must be either 'label' or 'idx'"
+            value = "label"
+        return self.label_vocab.set_index(key).to_dict()[value]
+
+    def __call__(self, predictions: Any, targets: Any, **kwargs) -> Dict:
+        """
+        Compute the metric based on the predictions and targets. Return a dictionary
+        of the results.
+        """
+        raise NotImplementedError("Inheriting classes must implement this function")
+
+
+class Top1Error(MetricFunction):
+    def __call__(
+        self, predictions: np.ndarray, targets: List, **kwargs
+    ) -> Dict[str, float]:
+        # Dictionary of labels and integer idx: {label -> idx}
+        label_vocab = self.label_vocab_as_dict(key="label")
+
+        # Compute the number of correct predictions
+        correct = 0
+        for i, prediction in enumerate(predictions):
+            predicted_class = np.argmax(prediction)
+            assert len(targets[i]) == 1
+            target_class = label_vocab[targets[i][0]]
+
+            if predicted_class == target_class:
+                correct += 1
+
+        error = correct / len(targets)
+        return {"top1_error": error}
```

Comment:
`sklearn.metrics` has a `top_k_accuracy_score`. We can use that `k==1`

Reply:
I like the idea of using the sklearn metric